### PR TITLE
[EDX-261] - New Footer Component

### PR DESCRIPTION
**Pull request by aralovelace**
> _NOTE TO REVIEWERS_ - please **do not review PRs in the `DRAFT` state**, as the PR may change substantially before it is ready to review. Thanks.

## Description

A PR description indicating the purpose of the PR.

* [EDX-261](https://ably.atlassian.net/browse/EDX-261?atlOrigin=eyJpIjoiZjA4YTJkZmZiMDMwNDg0MmJlZmYxYjNjNDIzMjI2MWYiLCJwIjoiaiJ9)

## Review

Instructions on how to review the PR. 

* Visit any pages
* then check the footer below should be the same as design - https://www.figma.com/file/vyr9ZPZ15eUWtCKSEktdEL/Docs-v1?node-id=1509%3A18097

### CODE REVIEW

**Review by MymmiJ — CHANGES_REQUESTED** [review]
Can you fix the rebase issue by copy-pasting textile files from main please?

File: src/components/Footer/Footer.tsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  `);
+
+  const menuDataItems = data.menuData.items;
+  const ablySocialLinks = data.ablySocial.items;
+  const bottomLinks = data.bottomMenu.items;
+
+  return (
+    <>
+      <footer className="bg-white antialiased" data-id="footer">
+        <div className="max-w-screen-xl mx-auto py-64 ui-grid-gap grid ui-grid-px grid-cols-6">
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:42]: This can be its own component

File: src/components/Footer/Footer.tsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+          <div className="col-span-full md:col-span-2 mb-30 md:mb-0">
+            <FooterLogoAndStatus />
+            <p className="mt-24 ui-text-p3">Connect with us on:</p>
+            <div className="pt-16 flex items-center">
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:46]: This can be its own component

File: src/components/Footer/Footer.tsx [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+              {ablySocialLinks
+                ? ablySocialLinks.map((social: { label: string; link: string }) => (
+                    <FooterSocial label={social.label} link={social.link} key={social.label} />
+                  ))
+                : null}
+            </div>
+          </div>
+          {menuDataItems
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:54]: This can be its own component

File: src/components/Footer/FooterLogoAndStatus.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import Logo from '@ably/ui/core/Logo';
+import FeaturedLink from '@ably/ui/core/FeaturedLink';
+import React from 'react';
+import { DOCUMENTATION_PATH } from '../../../data/transform/constants';
+
+export const FooterLogoAndStatus = () => (
```
Comment (MymmiJ) [inline src/components/Footer/FooterLogoAndStatus.tsx:6]: This is an odd combination of things; is there a reason these two are tied together?
If not, I would suggest making FooterStatus its own component and just rendering <Logo> and <FooterStatus> alongside each other
If there is a good reason, you should use that reason as the name of this component

File: src/components/Footer/FooterMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+export type FooterMenuItemType = { label: string; link: string };
+
+export const FooterMenu = ({ items, label }: { items: FooterMenuItemType[]; label: string }) => {
+  return (
+    <div className="col-span-full sm:col-span-3 md:col-span-1">
+      <h2 className="ui-footer-col-title">{label}</h2>
+      <ul>
+        {items
+          ? items.map((item: FooterMenuItemType) => (
+              <li className="p-menu-row-snug pb-8" key={item.label}>
```
Comment (MymmiJ) [inline src/components/Footer/FooterMenu.tsx:12]: This can be its own component

File: src/components/Footer/Footer.tsx [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+            ? menuDataItems.map((item: { items: FooterMenuItemType[]; label: string }) => (
+                <FooterMenu label={item.label} items={item.items} key={item.label} />
+              ))
+            : null}
+        </div>
+        <div className="max-w-screen-xl mx-auto py-40 border-t border-mid-grey w-full">
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:60]: This can be its own component

File: src/components/Footer/Footer.tsx [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+          <div className="inline-flex  justify-center w-full space-x-24">
+            {bottomLinks
+              ? bottomLinks.map((link: { label: string; link: string }) => (
+                  <a key={link.label} className="ui-footer-link" href={link.link}>
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:64]: This can be its own component

File: src/components/Footer/Footer.tsx [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+          label
+          link
+        }
+      }
+    }
+  `);
+
+  const menuDataItems = data.menuData.items;
+  const ablySocialLinks = data.ablySocial.items;
+  const bottomLinks = data.bottomMenu.items;
```
Comment (MymmiJ) [inline src/components/Footer/Footer.tsx:37]: if you do `data.bottomMenu.items ?? [];` for all of these, I think you won't have to check inline to see if the items exist before mapping later?

**Comment by aralovelace** [comment]
@MymmiJ  updated and convert it into smaller components

File: src/components/Footer/Footer.tsx [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    }
+  `);
+
+  const menuDataItems = data.menuData.items ?? [];
+  const ablySocialLinks = data.ablySocial.items ?? [];
+  const bottomLinks = data.bottomMenu.items ?? [];
+
+  return (
+    <>
+      <footer className="bg-white antialiased" data-id="footer">
```
[the remark this answers is not in the document]
Comment (aralovelace) [reply src/components/Footer/Footer.tsx]: @MymmiJ  just updated

**Review by MymmiJ — APPROVED** [review]
Approved, if there are more design changes to make from Srushtika please do make them but the code is good.

**Pull request merged**